### ISSUE TITLE
Action stops when credentials are not provided or are wrong

### ISSUE THREAD

**Issue by whoan** [whoan/docker-build-with-cache-action#28]

**Comment by whoan** [comment]
@leo-novu Can I see your workflow file? If you are using AWS ECR, since 4.3.0 now it's easier to login to the registry. I wonder if I introduced a breaking change in 4.2.0 or 4.3.0.
Since 4.2.0 you need to explicitly set `push_image_and_stages` to `false` if you don't provide the right credentials and don't expect the action to push the image and stages.

**Comment by leo-novu** [comment]
I dont believe the credentials are wrong.  In the previous actions that setup the credentials and login to ECR work.

---
name: CI
on: [pull_request]
jobs:
  Build_Docker_Image:
    name: Build Docker Image
    runs-on: ubuntu-latest
    env:
      ARTIFACTORY_CREDENTIALS: ${{ secrets.ARTIFACTORY_CREDENTIALS }}
    steps:
      - uses: actions/checkout@v2

      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1.3.2
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          aws-region: ${{ secrets.AWS_REGION }}

      - name: Login to Amazon ECR
        id: login-ecr
        uses: aws-actions/amazon-ecr-login@v1

      - name: Get AWS ECR credentials
        id: ecr-credentials
        run: |
          echo "::set-output name=username::AWS"
          echo "::set-output name=password::`aws ecr get-login-password`"

      - name: Docker build
        uses: whoan/docker-build-with-cache-action@v4.1.1
        with:
          username: ${{ steps.ecr-credentials.outputs.username }}
          password: ${{ steps.ecr-credentials.outputs.password }}
          registry: '822373129316.dkr.ecr.us-east-1.amazonaws.com'
          image_name: novu/ci-cache/hermesci
          build_extra_args: "--compress=true --build-arg=SKIP_TRANSLATIONS=true --build-arg=PRECOMPILE_ASSETS=0 --build-arg=ARTIFACTORY_CREDENTIALS=${{ secrets.ARTIFACTORY_CREDENTIALS }}"

![image](https://user-images.githubusercontent.com/40182906/79484957-15966f80-7fda-11ea-99b1-9b8c32917de1.png)

**Comment by whoan** [comment]
Could you please try with this @leo-novu ?:

```yml
- name: Docker build
  uses: whoan/docker-build-with-cache-action@v4.3
  with:
    username: ${{ secrets.AWS_ACCESS_KEY_ID }}
    password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
    registry: '822373129316.dkr.ecr.us-east-1.amazonaws.com'
    image_name: novu/ci-cache/hermesci
    build_extra_args: "--compress=true --build-arg=SKIP_TRANSLATIONS=true --build-arg=PRECOMPILE_ASSETS=0 --build-arg=ARTIFACTORY_CREDENTIALS=${{ secrets.ARTIFACTORY_CREDENTIALS }}"
```

**Comment by leo-novu** [comment]
Same results
```
Run whoan/docker-build-with-cache-action@v4.3.0
  with:
    username: AWS
    password: 
    registry: ********
    image_name: novu/ci-cache/novu-s3-utilsci
    build_extra_args: --compress=true --build-arg=SKIP_TRANSLATIONS=true --build-arg=PRECOMPILE_ASSETS=0 --build-arg=ARTIFACTORY_CREDENTIALS=***
    image_tag: latest
    context: .
    dockerfile: Dockerfile
    push_image_and_stages: true
    push_git_tag: false
    pull_image_and_stages: true
  env:
    ARTIFACTORY_CREDENTIALS: ***
    AWS_DEFAULT_REGION: ***
    AWS_REGION: ***
```

**Comment by whoan** [comment]
Don't know why I see this `username: AWS`, which seems wrong, but maybe is only how it is shown in the logs.
OK @leo-novu , I will take a closer look at it as soon as I can. Thanks for your tests.

**Comment by whoan** [comment]
@leo-novu Another question to track the problem. Didn't it work for you with v4.2.0?

**Comment by whoan** [comment]
I could not reproduce @leo-novu . Look at this run: https://github.com/whoan/docker-images/runs/594127068?check_suite_focus=true

The only way I could reproduce was setting wrong credentials, as you can see here: https://github.com/whoan/docker-images/runs/594068519?check_suite_focus=true

BTW, I set last version as v5.0.0 to avoid breaking existent workflows using v4. But I still wonder why it dodn't work for you in your last run.

Did you set the the correct keys? You need to set the ones you are currently setting in **Configure AWS Credentials** step, like this:

    username: ${{ secrets.AWS_ACCESS_KEY_ID }}
    password: ${{ secrets.AWS_SECRET_ACCESS_KEY }}

**Comment by whoan** [comment]
Closing this one as I can't reproduce. Feel free to open if you are sure there is something to fix in the action. Thanks.

**Comment by juniorb2ss** [comment]
@whoan 

I have same issue:

```
[Action Step] Log in to registry...
[Building and pushing a docker image to Amazon ECR/Building and pushing images to AWS ECR] Started container: 24513cd37183834996f4fc9555dec17da3c1e6d270fe2abb4958890764c7d56b
| Error response from daemon: login attempt to https://376372006699.dkr.ecr.us-east-1.amazonaws.com/v2/ failed with status: 401 Unauthorized
| INFO: Could not log in (please check credentials) - Won't be able to pull from private repos, nor to push to public/private repos
```

```yaml
name: Building and pushing a docker image to Amazon ECR

on:
  push:
    branches:         
    - master
env:
  SHA8: ${GITHUB_SHA::8-latest}

jobs:
  build:
    name: Building and pushing images to AWS ECR
    runs-on: ubuntu-latest
    steps:

    - name: Checkout
      uses: actions/checkout@master
      
    - name: Build/Pushing PHP image
      uses: whoan/docker-build-with-cache-action@v4
      with:
        username: "aws_key"
        password: "aws_secret"
        image_name: "image-name"
        context: .
        dockerfile: docker/php/base/Dockerfile
        push_image_and_stages: false
        push_git_tag: true
        image_tag: "${SHA8}"
        registry: 13243424.dkr.ecr.us-east-1.amazonaws.com
 ```

I'm testing using `act`.

**Comment by whoan** [comment]
Please use v5 @juniorb2ss

**Comment by whoan** [comment]
Fixed by #32
